internal/api/approval/state.go

```diff
@@ -87,41 +77,85 @@ func (s *state) addIgnoredReviewers(ignoredTeamMembers []string) {
+		if !rule.Fulfilled() {
+			return false
+		}
+	}
+
+	return true
+}
+
+func (s *state) pendingTeamNames() []string {
+	allPending := []string{}
+	for _, rule := range s.matchedRules {
+		rulePending := rule.PendingTeamNames()
+		allPending = uniqueAppend(allPending, rulePending)
+	}
+
+	return allPending
+}
+
+func (s *state) approvingTeamNames() []string {
+	allApproving := []string{}
+	for _, rule := range s.matchedRules {
+		ruleApproving := rule.ApprovingTeamNames()
+		allApproving = uniqueAppend(allApproving, ruleApproving)
+	}
+
+	return allApproving
+}
+
+func (s *state) shouldForceApprove() bool {
+	for _, rule := range s.matchedRules {
+		if rule.ConfigRule.ForceApproval {
+			return true
+		}
+	}
+
+	return false
+}
+
 func (s *state) result(log *log.Entry, teams []*github.Team) *Result {
 	result := &Result{
 		finalLabels:      s.labels,
 		ignoredReviewers: s.ignoredReviewers,
 	}
 
+	pendingTeamNames := s.pendingTeamNames()
+	approvingTeamNames := s.approvingTeamNames()
+	allRulesFulfilled := s.allRulesFulfilled()
+
 	// Compute the final status based on whether all required approvals have been met.
 	switch {
-	case s.rulesMatched == 0:
+	case len(s.matchedRules) == 0:
 		// No rules have been matched, which represents an error.
 		result.description = statusEventDescriptionNoRulesMatched
 		result.status = StatusEventStatusPending
 	case len(s.invalidTeamHandles) > 0:
 		// The configuration references a non-existent team
 		result.description = fmt.Sprintf(statusEventDescriptionInvalidTeamHandles, strings.Join(s.invalidTeamHandles, "\n"))
 		result.status = StatusEventStatusError
-	case s.forceApproval:
+	case s.shouldForceApprove():
 		// The PR is being forcibly approved.
 		result.description = statusEventDescriptionForciblyApproved
 		result.status = StatusEventStatusSuccess
-		result.reviewsToRequest = computeReviewsToRequest(log, teams, s.pendingTeamNames)
-	case len(s.pendingTeamNames) > 0:
+		result.reviewsToRequest = computeReviewsToRequest(log, teams, pendingTeamNames)
+	case len(pendingTeamNames) > 0 && !allRulesFulfilled:
 		// At least one team must still approve the PR before it goes green.
 		result.description = fmt.Sprintf(
-			statusEventDescriptionPendingFormatString, strings.Join(s.pendingTeamNames, "\n"))
+			statusEventDescriptionPendingFormatString, strings.Join(pendingTeamNames, "\n"))
 		result.status = StatusEventStatusPending
-		result.reviewsToRequest = computeReviewsToRequest(log, teams, s.pendingTeamNames)
-	case len(s.pendingTeamNames) == 0 && len(s.approvingTeamNames) == 0:
+		result.reviewsToRequest = computeReviewsToRequest(log, teams, pendingTeamNames)
+	case len(pendingTeamNames) == 0 && len(approvingTeamNames) == 0:
 		// No teams have been identified as having to be requested for a review.
 		// NOTE: This should not really happen in practice.
 		result.description = statusEventDescriptionNoReviewsRequested
 		result.status = StatusEventStatusSuccess
-	default:
```

Comment:
what could be the implications of removing `default`?

Reply:
Assuming no other cases match, I think we will ultimately log `failed to handle event` and the pull request status will never change. 

But I was a bit concerned about defaulting to `success` to be honest. I think I'd rather log error than assume the pull request is good to go.